src/components/input-field/input-field.e2e.ts

```diff
@@ -0,0 +1,379 @@
+                    'limel-input-field>>>.mdc-text-field__icon.trailing-icon>limel-icon'
+                );
+                expect(trailingIcon).toBeTruthy();
+                expect(trailingIcon).toEqualAttribute('name', 'external_link');
+            });
+        });
+
+        const validUrls = [
+            {
+                input: 'https://www.fullyqualified.com/',
+                expectedHref: 'https://www.fullyqualified.com/',
+            },
+            {
+                input: 'http://www.fullyqualified.com/',
+                expectedHref: 'http://www.fullyqualified.com/',
+            },
+            {
+                input: '//www.relativeprotocol.com',
+                expectedHref: '//www.relativeprotocol.com',
+            },
+            {
+                input: 'www.missingprotocol.com',
+                expectedHref: 'https://www.missingprotocol.com',
+            },
+            { input: 'nowww.com', expectedHref: 'https://nowww.com' },
+            { input: '/aRelativeUrl', expectedHref: '/aRelativeUrl' },
+            {
+                input:
+                    'nowww.com/with/email/address/in?queryString=someone@example.com',
+                expectedHref:
+                    'https://nowww.com/with/email/address/in?queryString=someone@example.com',
+            },
+        ];
+
+        validUrls.forEach((url) =>
+            describe(`with a value of '${url.input}'`, () => {
+                beforeEach(async () => {
+                    const nativeInput = await page.find(
+                        'limel-input-field>>>input'
+                    );
+                    nativeInput.focus();
+                    limelInput.setProperty('value', url.input);
+                    nativeInput.press('Tab');
+                    await page.waitForChanges();
+                });
+                it('is NOT considered invalid', () => {
+                    expect(inputContainer).not.toHaveClass(
+                        'mdc-text-field--invalid'
+                    );
+                });
+                it(`has a link with the href '${url.expectedHref}'`, async () => {
+                    const link = await page.find(
+                        'limel-input-field>>>.mdc-text-field__icon.trailing-icon'
+                    );
+                    expect(link).toEqualAttribute('href', url.expectedHref);
+                });
+                it('has a trailing icon indicating that the link can be opened', async () => {
+                    const icon = await page.find(
+                        'limel-input-field>>>.mdc-text-field__icon.trailing-icon>limel-icon'
+                    );
+                    expect(icon).toEqualAttribute('name', 'external_link');
+                });
+            })
+        );
+
+        const invalidUrls = [
+            { input: 'justoneword' },
+            { input: 'some words with spaces' },
+            // { input: 'an.email.address@example.com' }
```

Comment:
Ah, that's just cruft I forgot to remove. I wanted to test that it didn't accept an email address, but then realised that I would almost certainly not be able to make it _not_ incorrectly treat certain url:s as invalid because they contain an email address or something… So I decided to keep it simple.

Reply:
Removed!

⚡